### PR TITLE
Document form component validation error messages

### PR DESCRIPTION
**Pull request by jrbarnes9**
### What is the context of this PR?
Updated the following components and patterns to include error validation documentation and examples:

#### Components
- Access codes
    - Also renamed 'uac' pattern folders to 'access-code' *** **BREAKING CHANGE** ***
- Checkboxes
    - Also updated all examples to be more realistic use cases and use `legendIsPageTitle`
- Dates
- Durations
- Input
    - Also changed prefix/suffix background colour to match secondary buttons
    - Also updated width-constrained examples to use correct number classes
- Mutually exclusive
- Radios
- Select
- Summary
- Textarea
- Upload

#### Patterns
- Dates
- Download resources
- Durations
    - Also removed unnecessary time example
- Email addresses
- Feedback
- Find a support centre
- Numeric values
- Send a web form

Also fixed any typos in the content, and adjusted heading levels for consistency across similar templates.
 
### How to review
- Check all examples and error documentation for all the patterns and components listed above. 
- Check there are no issues with access codes after the folder name change

### CODE REVIEW

**Comment by jrbarnes9** [comment]
[Issue #1569: Upload component doesn't call `onsField` or `onsFieldset` so in turn, doesn't call `onsError`, so the validation pattern can't be used with it.](https://trello.com/c/dK0TQeYt/1220-issue-1569-upload-component-doesnt-call-onsfield-or-onsfieldset-so-in-turn-doesnt-call-onserror-so-the-validation-pattern-cant-b)

**Review by LauraC119 — APPROVED** [review]
all good from my point of view

File: src/components/search/index.njk [hunk 1]
```diff
@@ -23,9 +23,9 @@ Provide `accessiblePlaceholder: true` and a `label` to create an accessible plac
 
 A [width constraint](/components/input#width-constrained) class can be used to make the width of input smaller. If the search context is known, for instance a sequence of characters of a known size, the width constraint can help users.
 
-### Variants
+###
```
Comment (rmccar) [inline src/components/search/index.njk:26]: Missing title

File: src/components/summary/examples/summary-grouped-validation/index.njk [hunk 2]
```diff
@@ -29,7 +29,7 @@
                             },
                             {
                                 "rowTitle": "What was the value of the business's total sales of food?",
-                                "errorMessage": "Change one or more of the figures so they total the sum of £600",
+                                "errorMessage": "Change one or more of the figures so they sum to £600",
```
Comment (rmccar) [inline src/components/summary/examples/summary-grouped-validation/index.njk:32]: Text needs changing, not sure what its meant to say

File: src/patterns/feedback/examples/feedback-form-errors/index.njk [hunk 3]
```diff
@@ -81,6 +81,9 @@ layout: none
                             "description": "For example, questions not clear, answer options not relevant"
                         },
                         "value": "questions",
+                        "attributes": {
+                            "checked": true
+                        },
```
Comment (rmccar) [inline src/patterns/feedback/examples/feedback-form-errors/index.njk:86]:
```suggestion
                        "checked": true,
```
Attributes doesn't need to be used here

**Review by rmccar — COMMENTED** [review]
Ive noticed that all the error examples have the pattern error example *then* sample error messages, except for access codes. Maybe we should reverse this on access codes so it fits in with the rest. Also there is no error examples for the experimental patterns ("Find a support centre" and "Download resources")

File: src/components/summary/examples/summary-grouped-validation/index.njk (hunk 2 above)
Comment (jrbarnes9) [reply src/components/summary/examples/summary-grouped-validation/index.njk:32]: @LauraC119?

**Comment by jrbarnes9** [comment]
> Ive noticed that all the error examples have the pattern error example _then_ sample error messages, except for access codes. Maybe we should reverse this on access codes so it fits in with the rest. Also there is no error examples for the experimental patterns ("Find a support centre" and "Download resources")

I think it's better to show the example first, but some examples where it's a full page, I think are too long, so you'd miss the content.  This is why I flipped the order of these examples.

**Comment by rmccar** [comment]
> > Ive noticed that all the error examples have the pattern error example _then_ sample error messages, except for access codes. Maybe we should reverse this on access codes so it fits in with the rest. Also there is no error examples for the experimental patterns ("Find a support centre" and "Download resources")
> 
> I think it's better to show the example first, but some examples where it's a full page, I think are too long, so you'd miss the content. This is why I flipped the order of these examples.

Do we still not want examples for "Find a support centre" and "Download resources" though?

**Comment by jrbarnes9** [comment]
> > > Ive noticed that all the error examples have the pattern error example _then_ sample error messages, except for access codes. Maybe we should reverse this on access codes so it fits in with the rest. Also there is no error examples for the experimental patterns ("Find a support centre" and "Download resources")
> > 
> > 
> > I think it's better to show the example first, but some examples where it's a full page, I think are too long, so you'd miss the content. This is why I flipped the order of these examples.
> 
> Do we still not want examples for "Find a support centre" and "Download resources" though?

Yeah I'll add those now.

File: src/components/summary/examples/summary-grouped-validation/index.njk (hunk 2 above)
Comment (LauraC119) [reply src/components/summary/examples/summary-grouped-validation/index.njk:32]: should be "sum to £600". This has made me notice, though, that while the visual example is a sum of pounds the content on the page is a percentage. @jrbarnes9 do we normally match the example to the page content?

Comment (jrbarnes9) [reply src/components/summary/examples/summary-grouped-validation/index.njk:32]: Yeah usually.  I'll update this to match.

**Review by LauraC119 — APPROVED** [review]

**Review by boxadesign — APPROVED** [review]

**Pull request merged**